File: app/lib/features/analysis/domain/analysis.dart
class Analysis {
  const Analysis({
    required this.barcode,
    required this.score,
    required this.grade,
    required this.summary,
    required this.risks,
    required this.ingredients,
    this.createdAt,
  });

  final String barcode;
  final int score;
  final GradeLevel grade;
  final List<SummaryItem> summary;
  final List<Risk> risks;
  final List<Ingredient> ingredients;

  /// Local-only timestamp used for history sorting and display.
  /// Not part of the backend response; defaults to the moment of parsing.
  final DateTime? createdAt;
}

class SummaryItem {
  const SummaryItem({required this.message, this.code});

  final String message;
  final String? code;
}

class Risk {
  const Risk({
    required this.title,
    required this.severity,
    required this.description,
  });

  final String title;
  final RiskLevel severity;
  final String description;
}

enum RiskLevel { low, medium, high }

class Ingredient {
  const Ingredient({required this.name, required this.risk, this.description});

  final String name;
  final IngredientRiskLevel risk;
  final String? description;
}

enum IngredientRiskLevel { safe, caution, dangerous }

enum GradeLevel { excellent, good, average, poor }

File: backend/internal/models/analysis.go
package models

import "github.com/invopop/jsonschema"

type Analysis struct {
	Barcode     string        `json:"barcode"`
	Score       int           `json:"score"`
	Grade       GradeLevel    `json:"grade"`
	Summary     []SummaryItem `json:"summary"`
	Risks       []Risk        `json:"risks"`
	Ingredients []Ingredient  `json:"ingredients"`
}

type SummaryItem struct {
	Message string `json:"message"`
}

type Ingredient struct {
	Name        string `json:"name"`
	Risk        Risk   `json:"risk"`
	Description string `json:"description"`
}

type Risk struct {
	Severity    SeverityLevel `json:"severity"`
	Title       string        `json:"title"`
	Description string        `json:"description"`
}

type GradeLevel string

const (
	GradeExcellent GradeLevel = "excellent"
	GradeGood      GradeLevel = "good"
	GradeAverage   GradeLevel = "average"
	GradePoor      GradeLevel = "poor"
)

func (GradeLevel) JSONSchema() *jsonschema.Schema {
	return &jsonschema.Schema{
		Type: "string",
		Enum: []any{
			"excellent",
			"good",
			"average",
			"poor",
		},
	}
}

type SeverityLevel string

const (
	SeverityLow    SeverityLevel = "low"
	SeverityMedium SeverityLevel = "medium"
	SeverityHigh   SeverityLevel = "high"
)

func (SeverityLevel) JSONSchema() *jsonschema.Schema {
	return &jsonschema.Schema{
		Type: "string",
		Enum: []any{
			"low",
			"medium",
			"high",
		},
	}
}
